Installation/CMakeLists.txt

```diff
@@ -39,6 +39,27 @@ option(CGAL_HEADER_ONLY
 
 message("== Setting paths ==")
 
+if (NOT CGAL_DATA_DIR)
+  if(DEFINED ENV{CGAL_DATA_DIR})
+    set(CGAL_DATA_DIR $ENV{CGAL_DATA_DIR})
+  else()
+    if (EXISTS "${CMAKE_SOURCE_DIR}/data")
+      set(CGAL_DATA_DIR "${CMAKE_SOURCE_DIR}/data")
+    else()
+      if (EXISTS "${CMAKE_SOURCE_DIR}/Data/data")
+        set(CGAL_DATA_DIR "${CMAKE_SOURCE_DIR}/Data/data")
+      else()
+        message("CMAKE_SOURCE_DIR = ${CMAKE_SOURCE_DIR}")
+        message(WARNING "CGAL_DATA_DIR cannot be deduced, set the variable CGAL_DATA_DIR to set the default value of CGAL::data_file_path()")
+      endif()
+    endif()
+  endif()
+endif()
+
+if ( NOT "${CGAL_DATA_DIR}" STREQUAL "" )
+  add_definitions(-DCGAL_DATA_DIR="${CGAL_DATA_DIR}")
+endif()
```

Comment:
I have several problems with those few lines:
1. They are copy-pasted-modified at https://github.com/CGAL/cgal/pull/4229/files#diff-0b1d1a33526cb83c7915793ab2affe0fe4c7ab966a6b569fe1dbe6892d2e914bR322-R337 but I think only the version in `CGALConfig.cmake` should remain.
2. In the current pull-request, `CGAL_ROOT/Data/data` does not exist, and the feature cannot be tested. There should be one or two examples that actual uses `CGAL::data_file_path` in *this* PR, for the tests and review.
3. I do not like `add_definitions`. I would prefer `target_compile_definitions(CGAL INTERFACE ...)`. Otherwise that contaminates everything that includes `CGALConfig.cmake`, even user code that does not use CGAL.
4. I have added a directory `Data/data/` to my local repo to try and, as expected, that feature adds a compile option like `-DCGAL_DATA_DIR=\"/home/lrineau/Git/cgal-master/Data/data\"` to all the compilations using CGAL. Can there be a way to restrict that to actual CGAL examples and tests?

Reply:
About 1) when I did 3cefeeb that was also my guess that it was the only required change but if I remember correctly it was not sufficient and changes for ctest  and `Installation/CMakelist.txt` were still required (because they do do include the config file?). Sorry I don't remember the details.

Reply:
Thanks @lrineau  for the review.
2 and 3 are done.

About 4, I think we can keep this flag for all code that uses CGAL; it could be useful depending on the application ?
And even if this variable is unused, do you think is it a big deal to define an unused variable ?

Reply:
I find strange that any user code using CGAL and CMake will end up with that compilation flag `-DCGAL_DATA_DIR=\"/home/lrineau/Git/cgal-master/Data/data\"`.

Or worst, the CMake warning:
```
CMake Warning at Installation/lib/cmake/CGAL/CGALConfig.cmake:170 (message):
  CGAL_DATA_DIR cannot be deduced, set the variable CGAL_DATA_DIR to set the
  default value of CGAL::data_file_path()
```

Once CGAL is installed (using `make install`), the data files are meaningless and will not be installed, but there will still be that CMake warning.

... I confirm. I have install CGAL (from `master`) in `/home/lrineau/Git/cgal-master/build/install`, and then:
```
[lrineau@bonnard]~/Git/cgal-master/build/build-mesh_2% cmake -DCGAL_DIR=/home/lrineau/Git/cgal-master/build/install/lib64/cmake/CGAL ../../Mesh_2/examples/Mesh_2                       
-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/lib64/ccache/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/lib64/ccache/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using header-only CGAL
-- Targetting Unix Makefiles
-- Using /usr/lib64/ccache/c++ compiler.
-- Found GMP: /usr/lib64/libgmp.so  
-- Found MPFR: /usr/lib64/libmpfr.so  
-- Found Boost: /usr/include (found suitable version "1.69.0", minimum required is "1.48")  
-- Boost include dirs: /usr/include
-- Boost libraries:    
-- Using gcc version 4 or later. Adding -frounding-math
CMAKE_SOURCE_DIR = /home/lrineau/Git/cgal-master/build/install
CMake Warning at /home/lrineau/Git/cgal-master/build/install/lib64/cmake/CGAL/CGALConfig.cmake:170 (message):
  CGAL_DATA_DIR cannot be deduced, set the variable CGAL_DATA_DIR to set the
  default value of CGAL::data_file_path()
Call Stack (most recent call first):
  CMakeLists.txt:4 (find_package)

```

Reply:
Note that above I suggested to have this variable set in the macro `create_single_source_cgal_program()` (instead of in the config flag). @gdamiand said that he was worried for examples not using the macro so I picked the config file option. Note that the warning can also simply removed, it was here mainly for debug purpose anyway (it was an error in the first place before I turned it into a warning).

Reply:
> Note that above I suggested to have this variable set in the macro `create_single_source_cgal_program()` (instead of in the config flag).

That sounds better. There could also be a special target `CGAL::Data` for the examples not using the function `create_single_source_cgal_program`. @gdamiand, what do you think of that possibility? That would solve my point 1 (about the logic being copy-pasted in several places0, and point 4 (because only CGAL examples and tests using `create_single_source_cgal_program` or `CGAL::Data` would be impacted).

Reply:
Ok with this solution: CGAL::Data target and modify the create_single_source_cgal_program function.

Reply:
@lrineau and @sloriot Done in https://github.com/CGAL/cgal/pull/4229/commits/d140e5b89dbc0541d3e6a938744b6d620ad42214; need more tests.